src/main/java/ch/njol/skript/Skript.java

```diff
@@ -679,22 +675,63 @@ protected void afterErrors() {
 								TestTracker.testStarted("run scripts");
 								TestTracker.testFailed("exception was thrown during execution");
 							}
-							String results = new Gson().toJson(TestTracker.collectResults());
-							try {
-								Files.write(TestMode.RESULTS_FILE, results.getBytes(StandardCharsets.UTF_8));
-							} catch (IOException e) {
-								Skript.exception(e, "Failed to write test results.");
+							if (TestMode.JUNIT) {
+								SkriptLogger.setVerbosity(Verbosity.DEBUG);
+								info("Running all JUnit tests...");
+								long milliseconds = 0, tests = 0, fails = 0, ignored = 0, size = 0;
+								try {
+									List<Class<?>> classes = Lists.newArrayList(Utils.getClasses("org.skriptlang.skript.test", "tests"));
+									// Test that requires package access.
+									classes.add(Class.forName("ch.njol.skript.variables.FlatFileStorageTest"));
+									size = classes.size();
+									for (Class<?> clazz : classes) {
+										String test = clazz.getName();
+										SkriptJUnitTest.lastJUnitTest = test;
+										info("Running JUnit test '" + test + "'");
+										Result junit = JUnitCore.runClasses(clazz);
+										TestTracker.testStarted("JUnit: '" + test + "'");
+										shutdown += SkriptJUnitTest.getDelay();
+										SkriptJUnitTest.setDelay(0);
+										tests += junit.getRunCount();
+										milliseconds += junit.getRunTime();
+										ignored += junit.getIgnoreCount();
+										if (junit.getFailureCount() > 0) {
```

Comment:
Not needed

Reply:
Please explain better

Reply:
True, but it currently would only do one execution being the condition, versus a mathematical evaluation and an empty lambda check.

Reply:
The JIT already optimizes this and even if it didn't that would be about 4ns lost (4 billionths of a second). The compiler and JIT are pretty smart so the if statement makes no functional difference except for increasing the scope by one, reducing code readability.